leoism/step/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,124 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> slotsOpen = new ArrayList<TimeRange>();
+
+    // A requested meeting duration longer than the time of day should not be possible.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return slotsOpen;
+    }
+
+    if (events.isEmpty()) {
+      slotsOpen.add(TimeRange.WHOLE_DAY);
+      return slotsOpen;
+    }
+
+    ArrayList<TimeRange> slotsTaken = getSlotsTaken(events, request);
```

Comment:
I think that's an assumption we have to make since the only file we should modify is FindMeetingQuery.java in the first and second half and FindMeetingQueryTest.java in the second half of the walkthrough.

Reply:
If it's an assumption you're making please make a note of it in the comments (unless I missed that). Make sure it's in this function and on the comments for getSlotsTaken.